tests/run_all_tests.py

```diff
@@ -486,12 +486,36 @@ def create_and_run_test_obj(self, test_queries_obj_list, expected_res):
             elif self.action == 'override_expected_output':
                 test_obj.update_expected_output()
 
+    def _test_file_matches_category_output_tests(self, test_file):
+        # output tests run on files under fw_rules_tests and output_configs_tests dirs
+        if self.category == '':
+            return True
+        file_name = os.path.basename(test_file)
+        if file_name.startswith(self.category):
+            return True
+        if self.category == 'k8s' and not file_name.startswith(('calico', 'istio')):
+            return True
+        return False
+
+    def _test_file_matches_category_general_tests(self, test_file):
+        if self.category == '':
+            return True
+        if self.category + '_testcases' in test_file:
+            return True
+        if '_testcases' not in test_file:
+            return self._test_file_matches_category_output_tests(test_file)
```

Comment:
why matching for output tests within the function for general tests on this case?

Reply:
general test run on all the subdirs of tests/ ,
which include <category>_testcases directories, each belong to 1 category. And the remaining directories ( tests/fw_rules_tests and tests/output_config_tests) which are also the directories that "output" test runs on, and contain scheme_files from all categories